OsuApiClient/OsuClient.cs

```diff
@@ -47,6 +47,9 @@ public void Dispose()
         _disposed = true;
     }
 
+    public OsuCredentials SetCredentials(OsuCredentials credentials) =>
```

Comment:
I think you can use `public OsuCredentials? Credentials { private get; set; }` instead.